test/testdata/rbi/data.rb

```diff
@@ -0,0 +1,79 @@
+# typed: true
+
+class Define
+  Data.define # error: Not enough arguments provided for method `Data.define`. Expected: `1+`, got: `0`
+  Data.define(:x, :y)
+  Data.define("x", "y")
+
+  Data.define(:x) do
+    # Takes a block
+  end
+end
+
+Point = Data.define(:x, :y)
+
+class ValidInitializers
+  T.assert_type!(Point, T.class_of(Point))
+
+  point = Point.new(1, 2)
+  point.x
+  point.y
+  T.assert_type!(point, Point)
+
+  point = Point.new(x: 1, y: 2)
+  point.x
+  point.y
+  T.assert_type!(point, Point)
+
+  point = Point[1, 2]
+  point = Point[x: 1, y: 2]
```

Comment:
We didn't handle this type of initializer in the writer because it's not handled for `Struct`, but we could try adding it in this or another PR if desired.